app/controllers/public_menus_controller.rb

```diff
@@ -0,0 +1,17 @@
+class PublicMenusController < ApplicationController
+  before_action :set_menu
+  
+  def show
+    @business = Business.find_by(business_url: params[:business_url])
+    @location = Location.find_by(location_url: params[:location_url])
+  end
```

Comment:
you could probably get the business & location in a `setup` method.  Maybe change method `set_menu` to `setup`, and get everything you need in there.